### PR TITLE
feat: remove package-lock.json (#65)

### LINKED ISSUE #65: Remove package-lock.json
### Description

I refer to https://github.com/sindresorhus/ama/issues/479#issuecomment-310661514

TL;DR: lockfiles for apps, not libraries - also less noise on git diffs.

### PR DESCRIPTION
**Pull request by jh3y**
### What does this PR do?
- add .npmrc file
- include lockfile in .gitignore
- remove `package-lock.json`

### Related issues
- Closes #65 

### Checklist
- [x] I have checked the [contributing document](../blob/master/CONTRIBUTING.md)
- [x] I have added or updated any relevant documentation
- [x] I have added or updated any relevant tests

### CODE REVIEW

**Comment by jackdclark** [comment]
This closes #65

File: README.md [hunk 1]
```diff
@@ -455,8 +455,8 @@ Thanks goes to these wonderful people ([emoji key](https://allcontributors.org/d
...
...equests">👀</a> <a href="https://github.com/nearform/graphql-hooks/commits?author=Joezo" title="Tests">⚠️</a></td><td align="center"><a href="http://simoneb.github.io"><img src="https://avatars1.githubusercontent.com/u/20181?v=4" width="100px;" alt="Simone Busoli"/><br /><sub><b>Simone Busoli</b></sub></a><br /><a href="#question-simoneb" title="Answering Questions">💬</a> <a href="https://github.com/nearform/graphql-hooks/issues?q=author%3Asimoneb" title="Bug reports">🐛</a> <a href="https://github.com/nearform/graphql-hooks/commits?author=simoneb" title="Documentation">📖</a></td><td align="center"><a href="https://jheytompkins.com"><img src="https://avatars1.githubusercontent.com/u/842246?v=4" width="100px;" alt="jhey tompkins"/><br /><sub><b>jhey tompkins</b></sub></a><br /><a href="https://github.com/nearform/graphql-hooks/issues?q=author%3Ajh3y" title="Bug reports">🐛</a> <a href="https://github.com/nearform/graphql-hooks/commits?author=jh3y" title="Code">💻</a> <a href="#content-jh3y" title="Content">🖋</a> <a href="https://github.com/nearform/graphql-hooks/commits?author=jh3y" title="Documentation">📖</a> <a href="#review-jh3y" title="Reviewed Pull Requests">👀</a></td></tr></table>
```
Comment (Joezo) [inline README.md]: This shouldn't be updated manually. Or did you use the cli? Looks like it's switched from markdown tables to `<table`>

Comment (jh3y) [reply README.md]: Yeah. I used the CLI.

Comment (Joezo) [reply README.md]: Cool! Strange it chose to convert the markup!

**Comment by jh3y** [comment]
@all-contributors please add @jh3y for test, question, bug, code, content

**Pull request merged**